pkgs/by-name/ax/ax25ms/package.nix

```diff
@@ -0,0 +1,60 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, autoconf
+, protobuf
+, pkg-config
+, grpc
+, libtool
+, which
+, automake
+, libax25
+}:
+
+stdenv.mkDerivation {
+  pname = "ax25ms";
+  version = "unstable-2024-04-28";
+
+  src = fetchFromGitHub {
+    owner = "ThomasHabets";
+    repo = "ax25ms";
+    rev = "c7d7213bb182e4b60f655c3f9f1bcb2b2440406b";
+    hash = "sha256-GljGJa44topJ6T0g5wuU8GTHLKzNmQqUl8/AR+pw2+I=";
+  };
+
+  buildInputs = [
+    protobuf
+    grpc
+    libax25
+  ];
+
+  nativeBuildInputs = [
+    which
+    pkg-config
+    autoconf
+    libtool
+    automake
+  ];
+
+  postInstall = ''
+    set +e
+    for binary_path in "$out/bin"/*; do
+      filename=$(basename "$binary_path")
+      mv "$binary_path" "$out/bin/ax25ms-$filename"
```

Comment:
❓ What is the purpose of prefixing all installed binaries with `ax25ms`?

Reply:
Because otherwise, you will have anonymous programs such as `uploader` and `downloader` on your path, which don't inform you of where they came, and are likely to have name conflicts with other programs on the PATH. You would not know where `downloader` and `uploader` from unless you inspect further with `which`. The same issue existed in https://github.com/NixOS/nixpkgs/pull/298737 where this was fixed also. It's definitely a good thing to do.

Reply:
Thank you for the context, that makes a lot of sense! I think a short comment for uninitiated folks would be very helpful here, e.g. `Prefix package programs with package name to indicate package provenance and avoid potential conflict of generic program names like "uploader" that may already exist in $PATH`